src/components/Modeler.vue

```diff
@@ -679,10 +679,21 @@ export default {
       diagram.bounds.x = (clientX - paperOrigin.x) / scale.sx;
       diagram.bounds.y = (clientY - paperOrigin.y) / scale.sy;
 
+      if (this.isBoundaryEvent(definition)) {
+        this.setShapeCenterUnderCursor(diagram);
+      }
+
       // Our BPMN models are updated, now add to our nodes
       // @todo come up with random id
       this.addNode({ type: control.type, definition, diagram });
     },
+    isBoundaryEvent(definition) {
```

Comment:
Is there anyway we could modify the definition object to have this behaviour there? Do we control that object? If so it would be much preferable to having it here in the modeler. I would imaging the code would look something like:
`
isType(type) {
    return this.$type === type;
}
`

Reply:
That would be nice, but alas, we also don't own the definition object either; it comes from `bpmn-moddle`.

Reply:
Maybe we should consider ways to mitigate this then. One option is to wrap the definition object, which would also have the benefit of helping to separate us from the dependency. Another would be to extend it using prototypical inheritance, although that wouldn't help to isolate the dependency. What do you think? This behaviour really belongs in the definition...

Reply:
I'm thinking, we already have a sort of wrapper around these items: `node`. Currently, we treat node as a plain object with at least the following properties: `type`, `definition`, `diagram`.

It may be a good idea to wrap up the node functionality into a `Node` class. This node class could handle both returning its type, and functionality related to returning its centre position.

`definition` and `diagram` both come from `bpmn-moddle`, and both refer to the same entity that is eventually rendered on the paper.

Reply:
@chipit24 this sounds like a good plan! Could we begin this here by moving this behaviour (and the behaviour in my other comment) to the node?

Reply:
Yup :)